x-pack/plugins/security_solution/common/endpoint/data_loaders/index_endpoint_hearbeats.ts

```diff
@@ -40,9 +52,29 @@ export const indexEndpointHeartbeats = async (
         ingested,
       },
     };
+    // billable: true and missing billable are billed
+    if (i % 2) {
+      heartbeatDoc.billable = true;
+    }
+    return heartbeatDoc;
   });
 
-  const operations = docs.flatMap((doc) => [
+  // billable: false are not billed
+  const invalidDocs: EndpointHeartbeat[] = [
```

Comment:
this is now an additional doc to the number of `count` documents. i'd suggest to include it in `count`, or add a new option like `countOfInvalidDocuments` next to `count`

Reply:
This doesn't add to `count` since it is filtered out by the query. We want the count to not include it so we validate that invalid docs aren't incorrectly added to the count. The actual number of invalid docs does not matter here.

Reply:
i understand it doesn't matter, it's just violating the contract proposed by the function `indexEndpointHeartbeat(count)`, it looks like an unwanted side effect of the loader